torch/distributed/distributed_c10d.py

```diff
@@ -753,6 +784,61 @@ def recv(tensor,
         return src
 
 
+@contextlib.contextmanager
+def _group_manager(backend):
+    if backend == Backend.NCCL:
+        ProcessGroupNCCL.group_start()
+    try:
+        yield
+    finally:
+        if backend == Backend.NCCL:
+            ProcessGroupNCCL.group_end()
+
+
+def batch_isend_irecv(op_list,
+                      tensor_list,
+                      peer_list,
+                      group_list=None,
+                      tag_list=None):
+    """
+    Send or Receive tensors asynchronously and return a list of requests.
+
+    Process the first item in each passed parameters, and then the second item
+    in each passed parameters, etc.
+
+    Arguments:
+        op_list: list of point-to-point operations(type of each operations is either
+        ``torch.distributed.isend`` or ``torch.distributed.irecv``.
+        tensor_list (list[Tensor]): list of send or recv tensors.
+        peer_list (list[int]): list of peer ranks to send to or receive from.
+        group_list (list[ProcessGroup], Optional): list of groups to operator on.
```

Comment:
What is the use case of having a `group_list` here instead of just having a single group? For ProcessGroupNCCL, as per NCCL's guidance we can't use multiple process groups anyways.